querier.hh

```diff
@@ -65,29 +72,24 @@ public:
 protected:
     schema_ptr _schema;
     reader_permit _permit;
-    lw_shared_ptr<const dht::partition_range> _range;
-    std::unique_ptr<const query::partition_slice> _slice;
+    querier_reader_context_ptr _ctx;
     std::variant<flat_mutation_reader_v2, reader_concurrency_semaphore::inactive_read_handle> _reader;
```

Comment:
We could move all of the querier into an allocated object. Then moving queriers around is just moving unique_ptrs.

We could be even crazier and move all the contents into reader_permit (and then querier becomes a small wrapper around it). But we'll have many more reader_permit:s than queriers.